packages/utils/src/dappkit.ts

```diff
@@ -150,27 +152,40 @@ export function serializeDappKitRequestDeeplink(request: DappKitRequest) {
 
 // TODO: parsing query params yields broad types
 // once interface stabilizes, properly type the parsing
-export function parseDappkitResponseDepplink(url: string): DappKitResponse {
+export function parseDappkitResponseDepplink(url: string): [DappKitResponse, string] {
```

Comment:
You've been writing too much solidity 😄. Why not an object/nested object with names instead of an array?

Reply:
I would say I like Haskell too much :P 

I think Tuples are technically speaking more performant, but would agree that they are maybe not as ergonomic